utils/helper.py

```diff
@@ -65,9 +69,27 @@ def init_wallet(web3):
 
 
 def init_sgx_wallet(web3):
+    wait_for_node_config_file()
+    wait_for_sgx_key_name()
+    with open(NODE_CONFIG_FILEPATH, encoding='utf-8') as data_file:
+        config = json.loads(data_file.read())
+    return SgxWallet(os.environ['SGX_SERVER_URL'], web3, config['sgx_key_name'])
+
+
+def wait_for_node_config_file():
     while not os.path.isfile(NODE_CONFIG_FILEPATH):
         logger.info(f'Waiting for the {NODE_CONFIG_FILEPATH} to be created...')
         sleep(DEFAULT_SLEEP_TIMEOUT)
+
+
+def wait_for_sgx_key_name():
```

Comment:
I think we need to add timeout or limited number of iterations here

Reply:
@badrogger yeah, make sense.
but we don't know how long we should wait for the key.